### ISSUE TITLE
The most recent polyfills via polyfill-service?

### ISSUE THREAD

**Issue by dy** [azproduction/autopolyfiller#18]
Hi. It is really great tool, but it too fast becomes outdated due to lack of modern polyfills.

I know, there is an option to set up autopolyfiller to attach custom polyfills, but it is a separate headache for module developers. Instead of that it is faster to manually pick needed polyfills and concat them to the bundle. That makes autopolyfiller useless, unfortunately.

I’d suggest using the API of the most actual [polyfill database](https://github.com/Financial-Times/polyfill-service), instead of autopolyfiller-stable. That way you get rid of polyfills maintenance headache and make users happy.

**Comment by danielberndt** [comment]
Seems like https://github.com/Financial-Times/polyfill-service/issues/233 is fixed now. Any updates on this issue here?

**Comment by monolithed** [comment]
I see the following problems:

https://github.com/Financial-Times/polyfill-service/issues/331
https://github.com/Financial-Times/polyfill-service/issues/271
https://github.com/Financial-Times/polyfill-service/issues/270
https://github.com/Financial-Times/polyfill-service/issues/303

**Comment by sheo13666** [comment]
@azproduction Will the migration to polyfill-service?

**Comment by azproduction** [comment]
@sheo13666 I don't know about quality, but `polyfill-service` is less stable than it was originally :-) It is better to wait at least for separate polyfills repo.

**Comment by sheo13666** [comment]
@azproduction Used polyfill-service, stability problems did not notice

**Comment by smacker** [comment]
@azproduction some of current polyfills are broken. For example Array.prototype.some. But polyfill-service fixed it.

**Comment by triblondon** [comment]
@azproduction I'm sorry the polyfill service hasn't been easy to pick up for your use case.  We'd like to ensure it is, and we're working through those issues.

**Comment by azproduction** [comment]
@triblondon thanks. 
The biggest issue is that `polyfill-service` is not only database, but tool, which requires lots of unnecessary dependences. Any plans for extracting it as a standalone database?

**Comment by triblondon** [comment]
No. There's a discussion of that here: https://github.com/Financial-Times/polyfill-service/issues/125.  Basically the tool is required to test the polyfills, so the two are inseparable.  Also, since that discussion, we now have some polyfills that require dependencies that are private to the service like `_enqueueMicrotask`.  We are not intending to make the consumption of the raw polyfill source a supported use case, so for your purposes you would use the service API to generate the polyfill source that you need.

**Comment by azproduction** [comment]
> you would use the service API to generate the polyfill source that you need

@triblondon `getPolyfillString(options)` relies on `uaString` which is not possible to get in my case. 

Or can I hack it like that?
```js
getPolyfillString({
    uaString: '',
    features: {
        'Array.prorortpe.forEach': {}
    }
});
```

**Comment by triblondon** [comment]
The problem with doing that is that some polyfills have variants per-browser.  If a polyfill has totally different code in IE vs FF, and you don't specify a UA, we have no idea which variant to serve.  Some have default versions, but those tend not to work in the specific browsers that we have created variants for.

The library is designed to create the smallest possible bundle for a known browser.

That said I just had a thought for how we could support the best of both worlds, but it would be a major change.  I've logged it as https://github.com/Financial-Times/polyfill-service/issues/366

**Comment by azproduction** [comment]
@triblondon I see. And for Autopolyfiller browser-agnostic(or with runtime activation) polyfills are the must.

**Comment by triblondon** [comment]
Your current mechanism requires browser agnostic bundles, but I would argue that you could improve your tool by outputting a `<script>` tag targeting cdn.polyfill.io rather than offering up a single precompiled bundle that is served to all browsers. Then we wouldn't need to figure out a way of supporting browser agnostic bundles, which I see as an inferior method of using the service to the one we already have.

**Comment by jonathantneal** [comment]
Has this been resolved? It think the polyfill service provides the tools you need. I would love to use this today with grunt-autopolyfiller.

It finally dawned on me what all of this was doing, as it has now affected me :)

This is still a significant problem for JS devs who want something like autoprefixer for writing JS. Who don’t want to know all the ins and outs of old browsers, who want to only fetch the polyfills they need for a certain stack of browsers, but whose clients won’t necessarily let them link to an awesome cdn service as FT provides.

**Comment by dfcreative** [comment]
It is still a very requested feature in many respects.

For example, there is a [polyify](https://github.com/solomonhawk/polyify) transform, which can solve the issue of automatic polyfillying of any browserify bundles. That would remove a pain of polyfilling bundles for IE, mozilla, safari, iOS etc., as it is always a headache to manually search for features in code needed to polyfill and then generate proper URL for polyfill.io, like `
<script type="text/javascript" src="https://cdn.polyfill.io/v1/polyfill.js?features=default,Set,Element.prototype.matches"></script>`.

That would also remove issues related with [babelify](https://github.com/babel/babelify), to let it just easily use `Object.assign` instead of [`xtend`](https://www.npmjs.com/package/xtend) etc. For now it inserts [core-js](https://github.com/zloirock/core-js), quite huge and ungraceful.

@azproduction what problems are there preventing from closing that feature-request?

**Comment by triblondon** [comment]
Hi @azproduction, recent changes in the polyfill service may be of interest to you:

* We killed off variants. We could trivially make it possible to request a bundle without specifying a UA. If you are interested in picking up this project and that feature would make it possible for you to update autpolyfiller to use our new library, I'd be happy to implement it
* Version releases are now more frequent and should be less likely to violate semver. We have a deprecation process and better contribution standards

**Comment by jonathantneal** [comment]
If no one else picks this back up, I may. Please let me know if you are interested in helping. This would make an excellent Gulp/Grunt task at the least.

**Comment by dy** [comment]
@azproduction just wonder if you'd consider granting access to other maintainers: me, @jonathantneal, @browserify or alike.

**Comment by azproduction** [comment]
This project was abandoned for 4 years and has only 200 installations according to npm. 
Over the 4 years so many things changed. Has it any value today?

**Comment by dy** [comment]
@azproduction enabling polyfill-service as a source opens potential for browserify/rollup/webpack autopolyfillers. Definitely it has.
